### PR TITLE
Add support for multiple data directories

### PR DESCRIPTION
**Pull request by aoller**

### CODE REVIEW

**Comment by dfernandezramos** [comment]
👍

**Comment by ylatuya** [comment]
I think we need a helper method to resolve datadir paths since we will use it in several places.
Something like GetDataFilepath(string filename) that iterates over all dirs and throws an IO exception of the file is not found

File: VAS.Core/App.cs [hunk 1]
```diff
@@ -37,6 +37,11 @@ public abstract class App
 			set;
 		}
 
+		public List<string> DataDir {
```
Comment (ylatuya) [inline VAS.Core/App.cs]: This property should be below the static method with the other properties.

**Comment by ylatuya** [comment]
There is already a function RelativeToPrefix (string relativePath) to resolve paths relative to the prefix, we could follow the same scheme to resolve paths relative to the data dir

File: VAS.Core/App.cs [hunk 2]
```diff
@@ -55,7 +60,8 @@ public static void Init (App appInit, string evUninstalled, string softwareName,
 
 			if (Environment.GetEnvironmentVariable (evUninstalled) != null) {
 				App.Current.baseDirectory = Path.GetFullPath (".");
-				App.Current.dataDir = "../data";
+				App.Current.DataDir.Clear ();
```
Comment (ylatuya) [inline VAS.Core/App.cs]: Clear() is called in both cases. Why not leaving the DataDir unitialized, and initialize it here correctly? I prefer the DataDir being null rather to initialize to a value that's useless like "."

File: VAS.Core/App.cs [hunk 3]
```diff
@@ -25,18 +26,31 @@ public abstract class App
...
 		public string baseDirectory = ".";
 		public string configDirectory = ".";
-		public string dataDir = ".";
 
 		/* State */
 		public EventsAggregator EventsAggregator;
 
 		protected StyleConf style;
 
+		public List<string> DataDir {
```
Comment (dfernandezramos) [inline VAS.Core/App.cs]: As Andoni said, this should be below the static method with the other properties :)

File: VAS.Core/App.cs [hunk 4]
```diff
@@ -55,7 +69,7 @@ public static void Init (App appInit, string evUninstalled, string softwareName,
 
 			if (Environment.GetEnvironmentVariable (evUninstalled) != null) {
 				App.Current.baseDirectory = Path.GetFullPath (".");
-				App.Current.dataDir = "../data";
+				App.Current.DataDir.Add (appInit.RelativeToPrefix ("../data"));
```
Comment (dfernandezramos) [inline VAS.Core/App.cs]: this should be Current instead appInit

File: VAS.Core/App.cs [hunk 5]
```diff
@@ -69,7 +83,7 @@ public static void Init (App appInit, string evUninstalled, string softwareName,
 				}
 				if (!Directory.Exists (Path.Combine (App.Current.baseDirectory, "share", softwareName)))
 					Log.Warning ("Prefix directory not found");
-				App.Current.dataDir = Path.Combine (App.Current.baseDirectory, "share", softwareName.ToLower ());
+				App.Current.DataDir.Add (appInit.RelativeToPrefix (Path.Combine ("share", softwareName.ToLower ())));
```
Comment (dfernandezramos) [inline VAS.Core/App.cs]: this should be Current instead appInit

File: VAS.Core/App.cs (hunk 4 above)
Comment (dfernandezramos) [inline VAS.Core/App.cs]: It works with appInit too but to make it clearer, as all Init method works, we just use appInit to initialise Current. Then use Current for whatever you need :)

**Comment by dfernandezramos** [comment]
👍

**Pull request merged**